### PR TITLE
[WIP] Feature/warning on database num

### PR DESCRIPTION
**Pull request by lingtaolf**
Solving:  #56346

### Changelog category (leave one):
- New Feature

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
More warning message about number of tables , databases, parts, partitons will be added to system.metrics table.  

### Documentation entry for user-facing changes

- [x] Documentation is written (mandatory for new features)

> Information about CI checks: https://clickhouse.com/docs/en/development/continuous-integration/

### CODE REVIEW

**Comment by robot-ch-test-poll1** [comment]
*This is an automated comment for commit 19a4d886697a40a5a83a2254c581bcf13f176c6e with description of existing statuses. It's updated for the latest CI running*

[❌ Click here](https://s3.amazonaws.com/clickhouse-test-reports/57091/19a4d886697a40a5a83a2254c581bcf13f176c6e/ci_running.html) to open a full report in a separate page

<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>Docs Check</td><td>Builds and tests the documentation</td><td>❌ failure</td></tr>
<tbody>
</table>

File: src/Storages/MergeTree/MergeTreeData.cpp [hunk 1]
```diff
@@ -145,6 +151,7 @@ namespace ProfileEvents
 namespace CurrentMetrics
 {
     extern const Metric DelayedInserts;
+    extern const Metric AttachedDataParts;
```
Comment (alexey-milovidov) [inline src/Storages/MergeTree/MergeTreeData.cpp:154]: Unify this block with https://github.com/ClickHouse/ClickHouse/pull/57091/files#diff-b7d33a7abda5f3365d5e4fb58ca98b756f6401372b2e2d5eb449a2bd3444fb5eR127

**Comment by alexey-milovidov** [comment]
It's strange, but I'd like to write it differently. It should use less amount of code...

**Comment by alexey-milovidov** [comment]
Let the number of attached tables, databases, and parts be implemented as a CurrentMetric, held by the corresponding objects (IStorage, IDatabase, ...); by the way, it is already implemented for data parts.

The comparison with the max value should be implemented inside Context::getWarnings.

**Comment by alexey-milovidov** [comment]
The amount of code will be almost zero (less than 100 lines and about zero changes in interfaces).

**Comment by alexey-milovidov** [comment]
Interpreters and TablesLoader should not be modified.

**Comment by lingtaolf** [comment]
> Interpreters and TablesLoader should not be modified.

@alexey-milovidov Thx for reviewing, I will rewrite those code.

**Comment by lingtaolf** [comment]
@alexey-milovidov I created another PR #57375

**Pull request closed without merging**